applications/MappingApplication/custom_utilities/interface_condition.h

```diff
@@ -93,89 +89,206 @@ namespace Kratos
+                  && mNumPoints == 2) { // I am a linear line condition
+                  is_inside = MapperUtilities::ProjectPointToLine(mpCondition, rGlobalCoords,
                                                                   projection_local_coords,
-                                                                  projection_distance, m_normal);
-          } else if (m_geometry_family == GeometryData::Kratos_Quadrilateral) { // I am a quadrilateral condition
-              is_inside = MapperUtilities::ProjectPointToQuaddrilateral(m_p_condition, global_coords,
-                                                                        projection_local_coords,
-                                                                        projection_distance, m_normal);
-          } else {
-              KRATOS_ERROR << "MappingApplication; InterfaceCondition; \"EvaluateResult\" Unsupported Geometry!" << std::endl;
+                                                                  projection_distance);
+              } else if (mGeometryFamily == GeometryData::Kratos_Triangle 
+                        && mNumPoints == 3) { // I am a linear triangular condition
+                  is_inside = MapperUtilities::ProjectPointToTriangle(mpCondition, rGlobalCoords,
+                                                                      projection_local_coords,
+                                                                      projection_distance);
+              } else if (mGeometryFamily == GeometryData::Kratos_Quadrilateral
+                        && mNumPoints == 4) { // I am a linear quadrilateral condition
+                  is_inside = MapperUtilities::ProjectPointToQuadrilateral(mpCondition, rGlobalCoords,
+                                                                          projection_local_coords,
+                                                                          projection_distance);
+              } else if (mGeometryFamily == GeometryData::Kratos_Tetrahedra ||
+                         mGeometryFamily == GeometryData::Kratos_Prism ||
+                         mGeometryFamily == GeometryData::Kratos_Hexahedra) { // Volume Mapping
+                  is_inside = MapperUtilities::PointLocalCoordinatesInVolume(mpCondition, rGlobalCoords,
+                                                                             projection_local_coords,
+                                                                             projection_distance);
+              } else {
+                  std::cout << "MAPPER WARNING, Unsupported geometry, " 
+                            << "using an approximation" << std::endl;
+                  return false;
+              }
+          } catch (...) {
+              std::cout << "MAPPER WARNING, Functions necessary for projection "
+                        << "are not implemented in the core, " 
+                        << "using an approximation" << std::endl;
+              return false;
           }
```

Comment:
Yes it catches all exceptions.
I dont care about the reason it was not working because in this case I will not make use of any of the computed things and use an approximation instead (it returns false in the catch block). 
If I don't put it in a try catch block then the executions will terminate, so in my opinion "if" or "switch" statements won't work here

We can think about throwing different warnings in case of different exceptions such as the zero division you mentioned